.ci/release/Jenkinsfile

```diff
@@ -0,0 +1,235 @@
+        // Needs to append a line such as the following:
+        // 1.13.0: https://repo1.maven.org/maven2/co/elastic/apm/elastic-apm-agent/1.13.0/elastic-apm-agent-1.13.0.jar
+        stage('Update Cloudfoundry') {
+          steps {
+            dir("${basedir}") {
+              script {
+                sh("git checkout master")
+                def bareTag = sh(script: 'git tag|tail -1|sed s/v//', returnStdout: true)
+                def baseUrl = "https://repo1.maven.org/maven2/co/elastic/apm/elastic-apm-agent"
+                def lineToAppend = sh(script: "echo '$baseUrl: $baseUrl/$bareTag/elastic-apm-agent-$bareTag\.jar' >> cloudfoundry/index.yml ")
+                gitPush()
+              }
+            }
+          }
+        }
+        // 11. Wait for released package to be available in maven central
+        stage('Wait for artifact to be available in Maven Central') {
+          steps {
+            dir("${BASE_DIR}"){
+              script {
+                // We are looking for something like this:
+                // https://oss.sonatype.org/service/local/repositories/releases/content/co/elastic/apm/apm-agent-java/1.1.12/apm-agent-java-1.1.12.pom
+                def baseUrl = "https://oss.sonatype.org/service/local/repositories/releases/content/co/elastic/apm/apm-agent-java"
+                def bareTag = sh(script: 'git tag|tail -1|sed s/v//', returnStdout: true)
+                def fullUrl = sh(script: "$baseUrl/$bareTag/apm-agent-java-$bareTag\.pom")
+                waitUntil(initialRecurrencePeriod: 15000) {
+                  script {
+                    def ret = sh(script: "curl -fs $fullUrl >/dev/null 2>&1", returnStatus: true)
+                    echo "Waiting for the artifacts to be published on Sonatype"
+                    return ret == 0
+                  }
+                }
+              }
+            }
+
+          }
+        }
+        // 12. Publish release on Github. This will notify users watching repository.
+        stage('Publish release on GitHub') {
+          steps {
+            githubReleasePublish(id: ${env.RELEASE_ID})
+          }
+        }
+        // 13. Publish Docker images
+        stage('Docker push') {
+          when {
+            beforeAgent true
+            expression { return params.push_docker }
+          }
+          steps {
+            sh(label: "Build Docker image", script: "scripts/jenkins/build_docker.sh")
+            // Get Docker registry credentials
+            dockerLogin(secret: "${ELASTIC_DOCKER_SECRET}", registry: 'docker.elastic.co')
+            sh(label: "Push Docker image", script: "scripts/jenkins/push_docker.sh")
+          }
+        }
+        // 14. Opbeans
+        stage('Opbeans') {
+          environment {
+            REPO_NAME = "${OPBEANS_REPO}"
+          }
+          steps {
+            deleteDir()
+            dir("${OPBEANS_REPO}"){
+              git credentialsId: 'f6c7695a-671e-4f4f-a331-acdce44ff9ba',
```

Comment:
probably worth having a non-obscure credentials ID and/or a proper env variable for this.

Reply:
Using this method, we can use Jenkins secure credentials storage to ensure that we don't leak credentials to the log. Could you please elaborate on your reason for wanting this not to be obscured?

Reply:
There is no special reason, I was just surprised to not have something referred by a human-friendly name here.